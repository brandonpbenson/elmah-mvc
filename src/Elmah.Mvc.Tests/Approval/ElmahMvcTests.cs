using System.Text.RegularExpressions;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace Elmah.Mvc.Tests.Approval
{
    [UseReporter(typeof(DiffReporter))]
    public class ElmahMvcTests
    {
        private readonly string ElmahMvcAppUrl = "http://localhost:49800";

        [Test]
        public void should_open_elmah_page(string path)
        {
            // do
            var url = ElmahMvcAppUrl + path;

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_stylesheets()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/stylesheet";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));            
        }

        [Test]
        public void should_open_rss()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/rss";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));            
        }

        [Test]
        public void should_open_digest_rss()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/digestrss";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_details()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/detail?id=d5b2db50-0d5b-4c1d-a4dc-30324081191a";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_not_found_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/detail";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_json_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/json?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_xml_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/xml?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        private string GetContent(string url)
        {
            return HackServertTime(ApprovalTests.Asp.AspApprovals.GetUrlContents(url));
        }

        private string HackServertTime(string content)
        {
            return new Regex("Server time is \\d+:\\d+:\\d+").Replace(content, string.Empty);
        }
    }
}
