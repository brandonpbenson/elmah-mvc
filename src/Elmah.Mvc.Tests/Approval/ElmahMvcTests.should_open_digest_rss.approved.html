<rss xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" version="0.91">
  <channel>
    <title>Daily digest of errors in / on SEEKEY-NOTE</title>
    <link>http://localhost:49800/admin/elmah</link>
    <description>Daily digest of application errors</description>
    <language>en</language>
    <item>
      <title>Digest for 2012-06-08 (8 С‡РµСЂРІРЅСЏ 2012 СЂ.)</title>
      <description>&lt;ul&gt;
	&lt;li&gt;&lt;span title="System.FormatException"&gt;Format&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=f034b38b-a07a-4acd-b736-9decea0f707a"&gt;Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="System.FormatException"&gt;Format&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=e5a1477f-30f3-41d7-b9b2-d4d9b3751e4d"&gt;Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="System.FormatException"&gt;Format&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=52abe1f2-df17-42d1-b487-89600077f721"&gt;Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="Elmah.ApplicationException"&gt;Application&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7"&gt;Missing error identifier specification.&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="Elmah.ApplicationException"&gt;Application&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=1d13edd6-234a-488c-9d57-5fefb52f5a4b"&gt;Missing error identifier specification.&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="Elmah.ApplicationException"&gt;Application&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=3293aecd-e8f6-4cee-9d46-41fb060de9b1"&gt;Missing error identifier specification.&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="Elmah.ApplicationException"&gt;Application&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=a8d8e5be-0d48-4091-9445-5556dc0b7e61"&gt;Missing error identifier specification.&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="Elmah.ApplicationException"&gt;Application&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=4cb32a93-cb99-4517-9627-625fb2f74dc0"&gt;Missing error identifier specification.&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="Elmah.ApplicationException"&gt;Application&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=2072c375-6828-458a-8f6a-5817425c970a"&gt;Missing error identifier specification.&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="System.Web.HttpException"&gt;Http&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=d5b2db50-0d5b-4c1d-a4dc-30324081191a"&gt;A public action method &amp;#39;st&amp;#39; was not found on controller &amp;#39;Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahController&amp;#39;.&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="System.Web.HttpException"&gt;Http&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=79f1c6bf-e3e4-4c31-b0ab-7d99c95cd83a"&gt;A public action method &amp;#39;st&amp;#39; was not found on controller &amp;#39;Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahController&amp;#39;.&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="System.Web.HttpException"&gt;Http&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=24fa4f44-aed0-4c46-920e-e946f623d3ff"&gt;The controller for path &amp;#39;/not-found&amp;#39; was not found or does not implement IController.&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="System.Web.HttpException"&gt;Http&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=67b4e325-9342-4e71-8f83-9d700cb17557"&gt;The controller for path &amp;#39;/not-found&amp;#39; was not found or does not implement IController.&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;span title="System.NotImplementedException"&gt;NotImplemented&lt;/span&gt;: &lt;a href="http://localhost:49800/admin/elmah/detail?id=53e3c86e-85f8-4b00-b98c-2354808b15b8"&gt;The method or operation is not implemented.&lt;/a&gt;&lt;/li&gt;
&lt;/ul&gt;</description>
      <pubDate>Fri, 08 Jun 2012 15:39:34 GMT</pubDate>
    </item>
  </channel>
</rss>